baya/membership.py

```diff
@@ -174,13 +185,14 @@ def __init__(self, *operands):
                                 operand)
             # TODO: Check if operand is an instance of basestring and cast to
             # a BaseNode?
-        self._operands = operands
+        self._operands = tuple(operands)
+
+    def __hash__(self):
+        return hash((type(self), self._operands))
```

Comment:
This assumes that the dynamic roles methods will return the same thing if called repeatedly, which seems like a reasonable assumption for this class to make.